homeassistant/components/nws/sensor.py

```diff
@@ -81,37 +63,14 @@ def __init__(
         self.entity_description = description
 
         self._attr_name = f"{station} {description.name}"
-        if hass.config.units is US_CUSTOMARY_SYSTEM:
-            self._attr_native_unit_of_measurement = description.unit_convert
 
     @property
-    def native_value(self):
+    def native_value(self) -> StateType:
         """Return the state."""
-        value = self._nws.observation.get(self.entity_description.key)
-        if value is None:
-            return None
-        # Set alias to unit property -> prevent unnecessary hasattr calls
-        unit_of_measurement = self.native_unit_of_measurement
-        if unit_of_measurement == SPEED_MILES_PER_HOUR:
```

Comment:
We need a comment here, it might have gotten lost in some refactoring, but these conversions are needed because NWS provides inconsistent units in it's forecast and current observations.  homeassistant uses a single unit system for both.  The tests fail locally for me due to this.

Reply:
I suggest that you split your refactoring PR into three:
- add yourself to codeowners
- move SENSOR_TYPES from const.py to sensor.py, and replace unit constants with corresponding enums
- add missing type hints to config_flow.py and weather.py

Then in a follow-up, you can remove the conversion code. It will allow better focus on the "difficult to review" part.